### ISSUE TITLE
1.39.1 - Stopped accepting DNS traffic

### ISSUE THREAD

**Issue by henrybelk** [nextdns/nextdns#791]
**Context**
 - CLI Version: 1.39.1
 - Platform: Debian 11.6

&lt;Describe the bug>
Upgraded from 1.39.0 to 1.39.1. Both VMs stopped accepting DNS traffic. Reboot doesn't help. Shows running. cache-stats shows 'null'. 

 It looks like my config has been wiped out as well as I don't see the conditional settings I had in place. How can I roll back to 1.39.0?

**Comment by rs** [comment]
How did you upgrade?

What does `cat /etc/nextdns.conf` return?

**Comment by henrybelk** [comment]
The same as usual. `sudo apt update && sudo apt upgrade`

```
pi@DNS-01:~ $ cat /etc/nextdns.conf
detect-captive-portals false
hardened-privacy false
listen localhost:53
control /var/run/nextdns.sock
discovery-dns
max-ttl 0s
use-hosts true
auto-activate false
cache-max-age 0s
report-client-info false
timeout 5s
setup-router false
log-queries false
cache-size 0
mdns all
bogus-priv true
max-inflight-requests 256
```

**Comment by rs** [comment]
Do you have a backup of you previous config?

**Comment by cirrusflyer** [comment]
Upgrade broke both my installations and completely lost my complicated config file.  Great job folks. (I'm a paying customer).  Trying to start fresh and still broken.

**Comment by henrybelk** [comment]
Unfortunately not (unless it created a backup file before upgrading). I deleted my last snapshot of the VM yesterday as well. I can add back the custom config, but it is not responding on port 53. Now checking status, it shows not installed. After running install command, it shows running, but not responding on port 53 (PRTG monitor).

How can I remove and install 1.39.0? If I can get it back up, I can work on recreating my config.

```
pi@DNS-01:~ $ sudo nextdns status
{ []  [] [] [] false  0s 0s false   false false false 0s 0 false false}
not installed
pi@DNS-01:~ $ sudo nextdns install
{ [localhost:53] /var/run/nextdns.sock [] [] [] false 0 0s 0s false  all false true true 5s 256 false false}
NextDNS installed and started using systemd init
pi@DNS-01:~ $ sudo nextdns status
{ []  [] [] [] false  0s 0s false   false false false 0s 0 false false}
running
pi@DNS-01:~ $ sudo nextdns cache-stats
null
```

**Comment by rs** [comment]
Going back to 1.39.0 won't fix your issue, the problem is the loss of your config. Is this a router? Didn't you have `setup-router` on before? Without it, the default is to listen on localhost only.

**Comment by cirrusflyer** [comment]
Even a fresh installation on the same two linux PCs is giving me this same error:

{ []  [] [] [] false  0s 0s false   false false false 0s 0 false false}

**Comment by rs** [comment]
@cirrusflyer this is not an error but a leftover trace. It will go away with 1.39.2.

**Comment by cirrusflyer** [comment]
> @cirrusflyer this is not an error but a leftover trace. It will go away with 1.39.2.

Thanks.

**Comment by henrybelk** [comment]
Have two VMs running Raspberry Pi acting as DNS servers. Removed Nextdns from one and trying reinstall right now to 1.39.2. The other DNS server doesn't show 1.39.2 available as an upgrade yet.

**Comment by henrybelk** [comment]
Reinstalled on first one is accepting DNS traffic again. Will go through to configure custom settings. Install script recognized 1.39.2 to be installed, but it install 1.39.1. Was the issue that the custom config was wiped out that killed it?

```
pi@DNS-01:~ $ sh -c "$(curl -sL https://nextdns.io/install)"
INFO: OS: debian
INFO: GOARCH: amd64
INFO: GOOS: linux
INFO: NEXTDNS_BIN: /usr/bin/nextdns
INFO: INSTALL_RELEASE: 1.39.2
i) Install NextDNS
e) Exit
```

```
pi@DNS-01:~ $ sudo nextdns status
{ []  [] [] [] false  0s 0s false   false false false 0s 0 false false}
running
pi@DNS-01:~ $ sudo nextdns version
nextdns version 1.39.1
```

**Comment by rs** [comment]
The build is done. you can try to upgrade again, you should get the 1.39.2 now.

**Comment by henrybelk** [comment]
Able to configure split horizon and conditional profile ID and everything is working on the first DNS server. Still running and only shows 1.39.1 available. Will work on the second DNS server now.

Guess it is still processing on Github side.

```
Err:8 https://repo.nextdns.io/deb stable/main amd64 Packages
  File has unexpected size (422 != 427). Mirror sync in progress? [IP: 172.67.72.46 443]
  Hashes of expected file:
   - Filesize:427 [weak]
   - SHA512:e4a1d580ca3a5d4b1b0ed777a24c0c36e379130c7614b7d530fb449b496809aa49b8fa18cfcef734de0e63779ca91e148256072fd9eaf21f8fc112cf0f004ec8
   - SHA256:eecae6b7bfd0926efc92bb4175d0d5d359c6d91a5d47c8f54beb6176ceaf0207
   - SHA1:4fc44e27027bbd91424a781d58313bc7e45fa8ff [weak]
   - MD5Sum:60f530ac294dba51d77d379c3db93d8d [weak]
  Release file created at: Thu, 26 Jan 2023 17:36:27 +0000
Fetched 8,490 B in 1s (15.7 kB/s)
Reading package lists... Done
E: Failed to fetch https://repo.nextdns.io/deb/dists/stable/main/binary-amd64/Packages.gz  File has unexpected size (422 != 427). Mirror sync in progress? [IP: 172.67.72.46 443]
   Hashes of expected file:
    - Filesize:427 [weak]
    - SHA512:e4a1d580ca3a5d4b1b0ed777a24c0c36e379130c7614b7d530fb449b496809aa49b8fa18cfcef734de0e63779ca91e148256072fd9eaf21f8fc112cf0f004ec8
    - SHA256:eecae6b7bfd0926efc92bb4175d0d5d359c6d91a5d47c8f54beb6176ceaf0207
    - SHA1:4fc44e27027bbd91424a781d58313bc7e45fa8ff [weak]
    - MD5Sum:60f530ac294dba51d77d379c3db93d8d [weak]
   Release file created at: Thu, 26 Jan 2023 17:36:27 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by cirrusflyer** [comment]
Will the new version upgrade blast an existing config?  I have two more servers I need to upgrade.

**Comment by rs** [comment]
Found the issue, it will be fixed in 1.39.4. Please give it a few hours to make sure all it is fully tested before upgrading.
